### PR TITLE
Refactor server sandbox

### PR DESCRIPTION
**Pull request by 14rcole**
Make sandbox struct and it methods public, and add accessor methods for the struct fields

### CODE REVIEW

**Comment by mrunalp** [comment]
There are lint issues in this one.

**Comment by 14rcole** [comment]
@mheon PTAL

**Comment by mheon** [comment]
Code looks fine to me. This is necessary so we can use Sandboxes in the libkpod API.

File: server/sandbox.go [hunk 1]
```diff
@@ -162,35 +163,181 @@ var (
 	errSandboxClosedNetNS = errors.New("PodSandbox networking namespace is closed")
 )
 
-func (s *sandbox) addContainer(c *oci.Container) {
+// NewSandbox creates and populates a new pod sandbox
+// New sandboxes have no containers, no infra container, and no network namespaces associated with them
+// An infra container must be attached before the sandbox is added to the state
```
Comment (mrunalp) [inline server/sandbox.go]: nit: infra container isn't strictly necessary when using VMs.

Comment (rhatdan) [inline server/sandbox.go]: Lets get this merged and fix the nit in the next pull request.

**Comment by rhatdan** [comment]
@runcom PTAL

**Comment by mrunalp** [comment]
@sboeuf PTAL (we want to make sure that the refactor is fine for Clear Containers).

**Comment by runcom** [comment]
LGTM

**Comment by sboeuf** [comment]
@mrunalp the PR sounds good to me. I have just applied it on top of master to test with Clear Containers. I don't have any regression regarding the tests that were already passing.
LGTM

**Pull request merged**